### ISSUE TITLE
Demo fails with tensor size error

### ISSUE THREAD

**Issue by randerzander** [NVlabs/prismer#17]
When running `python demo.py --exp_name prismer_base`:
```
RuntimeError: The size of tensor a (3) must match the size of tensor b (9) at non-singleton dimension 0
```

The full trace is here: https://gist.github.com/randerzander/761d5d6b4a8aae09d773a4b1a5214ac7

**Comment by lorenmt** [comment]
Possibly the same problem reported here: https://github.com/NVlabs/prismer/issues/12

**Comment by randerzander** [comment]
I tried rebuilding the environment noted in #12, but have the same error.

In my case, I have: `Driver Version: 525.60.13    CUDA Version: 12.0 `

As I understand it, there is not yet a Torch package for CUDA 12, so I installed a version matching my toolkit version:
```
nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Fri_Dec_17_18:16:03_PST_2021
Cuda compilation tools, release 11.6, V11.6.55
Build cuda_11.6.r11.6/compiler.30794723_0
```

Setting up the python env:
```
conda create -n prismer python=3.10
conda activate prismer
git clone https://github.com/NVlabs/Prismer
cd Prismer
pip install torch --use-pep517 --extra-index-url https://download.pytorch.org/whl/cu116
conda install -c conda-forge ninja
pip install opencv-python
pip install -r requirements.txt
```
During the build of detectron2, I noticed errors about parameter pack expansion, and followed [this suggestion](https://github.com/NVIDIA/nccl/issues/650) to bump gcc/g++ down to 10.4.0.

After requirements installation:
```
python download_checkpoints.py --download_models=True --download_experts=True
cd experts/segmentation/mask2former/modeling/pixel_decoder/ops
sh make.sh
cd ~/projects/Prismer
```

And then running the demo as described in my initial post, I get the same error. I'm not sure what to try next.

**Comment by lorenmt** [comment]
This is so strange. I don’t think the gcc version should affect tensor shape anyway. Could you zip and send me your generated expert labels. I remember running with a different detrectron version also works well from my experience.

In the meantime, could you also track the input tensor shape, the output tensor shape from the vision encoder?

**Comment by lorenmt** [comment]
Resolved: https://github.com/NVlabs/prismer/commit/24376730ce47bd654efe790e552d43e31ccd1962

**Comment by randerzander** [comment]
> Resolved: [2437673](https://github.com/NVlabs/prismer/commit/24376730ce47bd654efe790e552d43e31ccd1962)

Thanks, that did fix it!
